src/node.c

```diff
@@ -522,8 +522,12 @@ node_get_base_url(nodePtr node)
 {
 	const gchar 	*baseUrl = NULL;
 
-	if (node->subscription)
+	if (node->subscription) {
 		baseUrl = subscription_get_homepage (node->subscription);
+		if (!baseUrl)
+			baseUrl = subscription_get_source (node->subscription);
```

Comment:
Are you sure it is good to use the subscription source directly?

Because it could be

     http://example.com/feed.atom

and would make a relative link work as

     http://example.com/feed.atom/somelink

instead of

     http://example.com/somelink

Reply:
You are right, I actually totally missed the leading "/" meaning append to domain.